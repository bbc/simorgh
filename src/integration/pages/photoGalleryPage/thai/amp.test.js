/**
 * @service thai
 * @pathname /thai/thailand-49950038
 */

import runAMPTests from '../ampTests';

describe('AMP', () => {
  describe(pageType, () => {
    runAMPTests(service);
  });
});
