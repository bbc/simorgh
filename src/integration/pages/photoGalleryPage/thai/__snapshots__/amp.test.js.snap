// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AMP Photo Gallery Page A11y I can see the skip to content link 1`] = `"ข้ามไปยังเนื้อหา"`;

exports[`AMP Photo Gallery Page AMP Consent config 1`] = `
{
  "consents": {
    "user-consent": {
      "promptIfUnknownForGeoGroup": "eea",
      "promptUI": "consent-prompt",
    },
  },
  "policy": {
    "default": {
      "timeout": {
        "fallbackAction": "reject",
        "seconds": 0,
      },
      "waitFor": {
        "user-consent": [],
      },
    },
  },
}
`;

exports[`AMP Photo Gallery Page AMP Geo config 1`] = `
{
  "AmpBind": true,
  "ISOCountryGroups": {
    "eea": [
      "at",
      "ax",
      "be",
      "bg",
      "bl",
      "cy",
      "cz",
      "de",
      "dk",
      "ea",
      "ee",
      "es",
      "fi",
      "fr",
      "gb",
      "gf",
      "gg",
      "gi",
      "gp",
      "gr",
      "hr",
      "hu",
      "ic",
      "ie",
      "im",
      "is",
      "it",
      "je",
      "li",
      "lt",
      "lu",
      "lv",
      "mf",
      "mq",
      "mt",
      "nc",
      "nl",
      "no",
      "pf",
      "pl",
      "pm",
      "pt",
      "re",
      "ro",
      "se",
      "si",
      "sj",
      "sk",
      "tf",
      "va",
      "wf",
      "yt",
    ],
    "gbOrUnknown": [
      "gb",
      "gg",
      "im",
      "je",
      "uk",
      "unknown",
    ],
  },
}
`;

exports[`AMP Photo Gallery Page Analytics ATI 1`] = `
{
  "requests": {
    "base": "https://logws1363.ati-host.net?",
    "pageview": "\${base}s=598343&s2=90&p=thai.thailand.photo_gallery.49950038.page&r=\${screenWidth}x\${screenHeight}x\${screenColorDepth}&re=\${availableScreenWidth}x\${availableScreenHeight}&hl=\${timestamp}&lng=\${browserLanguage}&x1=[urn%3Abbc%3Acps%3Acurie%3Aasset%3A5858e31f-36a4-ac44-ab69-9c340b45dd49]&x2=[amp]&x3=[news-thai]&x4=[th]&x5=[\${sourceUrl}]&x6=[\${documentReferrer}]&x7=[article-photo-gallery]&x8=[simorgh]&x9=[6%2520%E0%B8%95%E0%B8%B8%E0%B8%A5%E0%B8%B2%2520%3A%2520%E0%B8%98%E0%B8%A3%E0%B8%A3%E0%B8%A1%E0%B8%A8%E0%B8%B2%E0%B8%AA%E0%B8%95%E0%B8%A3%E0%B9%8C%E0%B8%88%E0%B8%B1%E0%B8%94%E0%B8%81%E0%B8%B4%E0%B8%88%E0%B8%81%E0%B8%A3%E0%B8%A3%E0%B8%A1%E0%B8%A3%E0%B8%B3%E0%B8%A5%E0%B8%B6%E0%B8%81%252043%2520%E0%B8%9B%E0%B8%B5%25206%2520%E0%B8%95%E0%B8%B8%E0%B8%A5%E0%B8%B2%25202519%2520-%2520BBC%2520News%2520%E0%B9%84%E0%B8%97%E0%B8%A2]&x11=[2019-10-06T05%3A14%3A53.000Z]&x12=[2019-10-06T05%3A14%3A53.000Z]&x16=[WS%20-%20Keep%20me%20on%20trend]&x17=[News]&ref=\${documentReferrer}",
  },
  "transport": {
    "beacon": false,
    "image": true,
    "xhrpost": false,
  },
  "triggers": {
    "trackPageview": {
      "on": "visible",
      "request": "pageview",
    },
  },
}
`;

exports[`AMP Photo Gallery Page Analytics chartbeat 1`] = `
{
  "vars": {
    "contentType": "PGL",
    "domain": "test.bbc.co.uk",
    "sections": "Thai, Thai - PGL",
    "title": "6 ตุลา : ธรรมศาสตร์จัดกิจกรรมรำลึก 43 ปี 6 ตุลา 2519",
    "uid": 50924,
    "virtualReferrer": "\${documentReferrer}",
  },
}
`;

exports[`AMP Photo Gallery Page Analytics comscore 1`] = `
{
  "extraUrlParams": {
    "comscorekw": "amp",
  },
  "vars": {
    "c2": "17986528",
  },
}
`;

exports[`AMP Photo Gallery Page Footer Anchors should match text and url 1`] = `
{
  "text": "BBC News, ไทย",
  "url": "/thai",
}
`;

exports[`AMP Photo Gallery Page Footer Anchors should match text and url 2`] = `
{
  "text": "ทำไมคุณจึงไว้วางใจ บีบีซี ได้",
  "url": "https://www.bbc.com/thai/institutional-49281839",
}
`;

exports[`AMP Photo Gallery Page Footer Anchors should match text and url 3`] = `
{
  "text": "เงื่อนไขการใช้งานของ บีบีซี",
  "url": "https://www.bbc.com/thai/institutional-38403477",
}
`;

exports[`AMP Photo Gallery Page Footer Anchors should match text and url 4`] = `
{
  "text": "เกี่ยวกับบีบีซี",
  "url": "https://www.bbc.com/thai/institutional-38403476",
}
`;

exports[`AMP Photo Gallery Page Footer Anchors should match text and url 5`] = `
{
  "text": "นโยบายความเป็นส่วนตัว",
  "url": "https://www.bbc.com/thai/institutional-38497681",
}
`;

exports[`AMP Photo Gallery Page Footer Anchors should match text and url 6`] = `
{
  "text": "คุกกีส์",
  "url": "https://www.bbc.com/usingthebbc/cookies/",
}
`;

exports[`AMP Photo Gallery Page Footer Anchors should match text and url 7`] = `
{
  "text": "ติดต่อบีบีซี",
  "url": "https://www.bbc.co.uk/thai/send/u50853797",
}
`;

exports[`AMP Photo Gallery Page Footer Anchors should match text and url 8`] = `
{
  "text": "อ่านเกี่ยวกับแนวทางของเราในการติดต่อกับลิงก์ภายนอก",
  "url": "https://www.bbc.co.uk/editorialguidelines/guidance/feeds-and-links",
}
`;

exports[`AMP Photo Gallery Page Footer Cookie Settings Button should match text and on handler 1`] = `
{
  "on": "tap:consent.prompt, privacy.hide, cookie.show, AMP.setState({ isManagingSettings: true })",
  "text": "Do not share or sell my info",
}
`;

exports[`AMP Photo Gallery Page Footer I can click on the BBC branding and it would take me to the homepage 1`] = `"/thai"`;

exports[`AMP Photo Gallery Page Footer Paragraphs should match text 1`] = `"© 2024 บีบีซี. บีบีซีไม่มีส่วนรับผิดชอบต่อเนื้อหาของเว็บไซต์ภายนอก. นโยบายของเราเรื่องการเชื่อมต่อไปยังลิงก์ภายนอก. อ่านเกี่ยวกับแนวทางของเราในการติดต่อกับลิงก์ภายนอก"`;

exports[`AMP Photo Gallery Page Header I can see the branding 1`] = `"BBC News, ไทย"`;

exports[`AMP Photo Gallery Page Header Navigation link should match text and url 1`] = `
{
  "text": "หน้าแรก",
  "url": "/thai",
}
`;

exports[`AMP Photo Gallery Page Header Navigation link should match text and url 2`] = `
{
  "text": "ประเทศไทย",
  "url": "/thai/topics/cjgn73g98rqt",
}
`;

exports[`AMP Photo Gallery Page Header Navigation link should match text and url 3`] = `
{
  "text": "ต่างประเทศ",
  "url": "/thai/topics/c5v124k8lj7t",
}
`;

exports[`AMP Photo Gallery Page Header Navigation link should match text and url 4`] = `
{
  "text": "วิทยาศาสตร์",
  "url": "/thai/topics/c5qvp1q33p0t",
}
`;

exports[`AMP Photo Gallery Page Header Navigation link should match text and url 5`] = `
{
  "text": "สุขภาพ",
  "url": "/thai/topics/cyx5kz25zxdt",
}
`;

exports[`AMP Photo Gallery Page Header Navigation link should match text and url 6`] = `
{
  "text": "วิดีโอ",
  "url": "/thai/topics/c5ljd3jng4nt",
}
`;

exports[`AMP Photo Gallery Page Header Navigation link should match text and url 7`] = `
{
  "text": "ยอดนิยม",
  "url": "/thai/popular/read",
}
`;

exports[`AMP Photo Gallery Page Image Caption should match text 1`] = `"คำบรรยายภาพ, ประติมานุสรณ์ ม. ธรรมศาสตร์ เป็นพื้นที่ความทรงจำในเหตุการณ์ 6 ตุลา 2519"`;

exports[`AMP Photo Gallery Page Image should match snapshot 1`] = `
<amp-img
  alt="ดอกไม้ถูกนำมาวางไว้ที่ประติมานุสรณ์"
  attribution="Thai news Pix"
  class="bbc-image"
  data-hero="true"
  fallback=""
  height="549"
  layout="responsive"
  src="https://ichef.bbci.co.uk/ace/ws/640/cpsprodpb/17431/production/_109118259_01.jpg"
  srcset="https://ichef.bbci.co.uk/ace/ws/240/cpsprodpb/17431/production/_109118259_01.jpg 240w, https://ichef.bbci.co.uk/ace/ws/320/cpsprodpb/17431/production/_109118259_01.jpg 320w, https://ichef.bbci.co.uk/ace/ws/480/cpsprodpb/17431/production/_109118259_01.jpg 480w, https://ichef.bbci.co.uk/ace/ws/624/cpsprodpb/17431/production/_109118259_01.jpg 624w, https://ichef.bbci.co.uk/ace/ws/800/cpsprodpb/17431/production/_109118259_01.jpg 800w"
  width="976"
/>
`;

exports[`AMP Photo Gallery Page Main heading should match text 1`] = `"6 ตุลา : ธรรมศาสตร์จัดกิจกรรมรำลึก 43 ปี 6 ตุลา 2519"`;

exports[`AMP Photo Gallery Page SEO Apple Touch Icon should match attributes 1`] = `
{
  "sizes": null,
  "url": "http://localhost:7080/thai/images/icons/icon-192x192.png",
}
`;

exports[`AMP Photo Gallery Page SEO Apple Touch Icon should match attributes 2`] = `
{
  "sizes": "72x72",
  "url": "https://static.files.bbci.co.uk/ws/simorgh-assets/public/thai/images/icons/icon-72x72.png",
}
`;

exports[`AMP Photo Gallery Page SEO Apple Touch Icon should match attributes 3`] = `
{
  "sizes": "96x96",
  "url": "https://static.files.bbci.co.uk/ws/simorgh-assets/public/thai/images/icons/icon-96x96.png",
}
`;

exports[`AMP Photo Gallery Page SEO Apple Touch Icon should match attributes 4`] = `
{
  "sizes": "128x128",
  "url": "https://static.files.bbci.co.uk/ws/simorgh-assets/public/thai/images/icons/icon-128x128.png",
}
`;

exports[`AMP Photo Gallery Page SEO Apple Touch Icon should match attributes 5`] = `
{
  "sizes": "144x144",
  "url": "https://static.files.bbci.co.uk/ws/simorgh-assets/public/thai/images/icons/icon-144x144.png",
}
`;

exports[`AMP Photo Gallery Page SEO Apple Touch Icon should match attributes 6`] = `
{
  "sizes": "152x152",
  "url": "https://static.files.bbci.co.uk/ws/simorgh-assets/public/thai/images/icons/icon-152x152.png",
}
`;

exports[`AMP Photo Gallery Page SEO Apple Touch Icon should match attributes 7`] = `
{
  "sizes": "180x180",
  "url": "https://static.files.bbci.co.uk/ws/simorgh-assets/public/thai/images/icons/icon-180x180.png",
}
`;

exports[`AMP Photo Gallery Page SEO Apple Touch Icon should match attributes 8`] = `
{
  "sizes": "192x192",
  "url": "https://static.files.bbci.co.uk/ws/simorgh-assets/public/thai/images/icons/icon-192x192.png",
}
`;

exports[`AMP Photo Gallery Page SEO Apple Touch Icon should match attributes 9`] = `
{
  "sizes": "384x384",
  "url": "https://static.files.bbci.co.uk/ws/simorgh-assets/public/thai/images/icons/icon-384x384.png",
}
`;

exports[`AMP Photo Gallery Page SEO Apple Touch Icon should match attributes 10`] = `
{
  "sizes": "512x512",
  "url": "https://static.files.bbci.co.uk/ws/simorgh-assets/public/thai/images/icons/icon-512x512.png",
}
`;

exports[`AMP Photo Gallery Page SEO Canonical link 1`] = `"http://localhost:7080/thai/thailand-49950038"`;

exports[`AMP Photo Gallery Page SEO Dir attribute 1`] = `"ltr"`;

exports[`AMP Photo Gallery Page SEO FB app ID 1`] = `"1609039196070050"`;

exports[`AMP Photo Gallery Page SEO Heading level 1 1`] = `"6 ตุลา : ธรรมศาสตร์จัดกิจกรรมรำลึก 43 ปี 6 ตุลา 2519"`;

exports[`AMP Photo Gallery Page SEO Lang attribute 1`] = `"th"`;

exports[`AMP Photo Gallery Page SEO Linked data should match text 1`] = `
{
  "@context": "http://schema.org",
  "@graph": [
    {
      "@type": "Article",
      "author": {
        "@type": "NewsMediaOrganization",
        "logo": {
          "@type": "ImageObject",
          "height": 576,
          "url": "https://news.files.bbci.co.uk/ws/img/logos/og/thai.png",
          "width": 1024,
        },
        "name": "BBC News ไทย",
        "noBylinesPolicy": "https://www.bbc.com/thai/institutional-49281839#authorexpertise",
      },
      "dateModified": "2019-10-06T05:14:53.000Z",
      "datePublished": "2019-10-06T05:14:53.000Z",
      "description": "มหาวิทยาลัยธรรมศาสตร์ ซึ่งมีสภาพไม่ต่างจาก “ลานประหาร” ในวันล้อมปราบนักศึกษาและประชาชน ได้จัดกิจกรรมรำลึก 43 ปี 6 ตุลา 2519 มีพิธีทำบุญตักบาตรพระสงฆ์จำนวน 19 รูป พิธีวางพวงมาลาและดอกไม้ ณ ประติมานุสรณ์ และกล่าวไว้อาลัยแด่ผู้เสียชีวิตในเหตุการณ์ พร้อมอ่านรายชื่อของทุกคนด้วย",
      "headline": "6 ตุลา : ธรรมศาสตร์จัดกิจกรรมรำลึก 43 ปี 6 ตุลา 2519",
      "image": {
        "@type": "ImageObject",
        "height": 576,
        "url": "https://ichef.bbci.co.uk/news/1024/branded_thai/17431/production/_109118259_01.jpg",
        "width": 1024,
      },
      "inLanguage": {
        "@type": "Language",
        "alternateName": "th",
        "name": "Thai",
      },
      "mainEntityOfPage": {
        "@id": "http://localhost:7080/thai/thailand-49950038",
        "@type": "WebPage",
        "name": "6 ตุลา : ธรรมศาสตร์จัดกิจกรรมรำลึก 43 ปี 6 ตุลา 2519",
      },
      "publisher": {
        "@type": "NewsMediaOrganization",
        "logo": {
          "@type": "ImageObject",
          "height": 576,
          "url": "https://news.files.bbci.co.uk/ws/img/logos/og/thai.png",
          "width": 1024,
        },
        "name": "BBC News ไทย",
        "publishingPrinciples": "https://www.bbc.com/thai/institutional-49281839",
      },
      "thumbnailUrl": "https://ichef.bbci.co.uk/news/1024/branded_thai/17431/production/_109118259_01.jpg",
      "url": "http://localhost:7080/thai/thailand-49950038",
    },
  ],
}
`;

exports[`AMP Photo Gallery Page SEO OG description 1`] = `"มหาวิทยาลัยธรรมศาสตร์ ซึ่งมีสภาพไม่ต่างจาก “ลานประหาร” ในวันล้อมปราบนักศึกษาและประชาชน ได้จัดกิจกรรมรำลึก 43 ปี 6 ตุลา 2519 มีพิธีทำบุญตักบาตรพระสงฆ์จำนวน 19 รูป พิธีวางพวงมาลาและดอกไม้ ณ ประติมานุสรณ์ และกล่าวไว้อาลัยแด่ผู้เสียชีวิตในเหตุการณ์ พร้อมอ่านรายชื่อของทุกคนด้วย"`;

exports[`AMP Photo Gallery Page SEO OG image 1`] = `"https://ichef.bbci.co.uk/news/1024/branded_thai/17431/production/_109118259_01.jpg"`;

exports[`AMP Photo Gallery Page SEO OG image alt 1`] = `"ดอกไม้ถูกนำมาวางไว้ที่ประติมานุสรณ์"`;

exports[`AMP Photo Gallery Page SEO OG locale 1`] = `"th-TH"`;

exports[`AMP Photo Gallery Page SEO OG site name 1`] = `"BBC News ไทย"`;

exports[`AMP Photo Gallery Page SEO OG title 1`] = `"6 ตุลา : ธรรมศาสตร์จัดกิจกรรมรำลึก 43 ปี 6 ตุลา 2519 - BBC News ไทย"`;

exports[`AMP Photo Gallery Page SEO OG type 1`] = `"article"`;

exports[`AMP Photo Gallery Page SEO OG url 1`] = `"http://localhost:7080/thai/thailand-49950038"`;

exports[`AMP Photo Gallery Page SEO Page title 1`] = `"6 ตุลา : ธรรมศาสตร์จัดกิจกรรมรำลึก 43 ปี 6 ตุลา 2519 - BBC News ไทย"`;

exports[`AMP Photo Gallery Page SEO Robots meta tag 1`] = `"noodp, noydir, max-image-preview:large"`;

exports[`AMP Photo Gallery Page SEO Twitter card 1`] = `"summary_large_image"`;

exports[`AMP Photo Gallery Page SEO Twitter creator 1`] = `"@bbc_thailand"`;

exports[`AMP Photo Gallery Page SEO Twitter description 1`] = `"มหาวิทยาลัยธรรมศาสตร์ ซึ่งมีสภาพไม่ต่างจาก “ลานประหาร” ในวันล้อมปราบนักศึกษาและประชาชน ได้จัดกิจกรรมรำลึก 43 ปี 6 ตุลา 2519 มีพิธีทำบุญตักบาตรพระสงฆ์จำนวน 19 รูป พิธีวางพวงมาลาและดอกไม้ ณ ประติมานุสรณ์ และกล่าวไว้อาลัยแด่ผู้เสียชีวิตในเหตุการณ์ พร้อมอ่านรายชื่อของทุกคนด้วย"`;

exports[`AMP Photo Gallery Page SEO Twitter image alt 1`] = `"ดอกไม้ถูกนำมาวางไว้ที่ประติมานุสรณ์"`;

exports[`AMP Photo Gallery Page SEO Twitter image src 1`] = `"https://ichef.bbci.co.uk/news/1024/branded_thai/17431/production/_109118259_01.jpg"`;

exports[`AMP Photo Gallery Page SEO Twitter site 1`] = `"@bbc_thailand"`;

exports[`AMP Photo Gallery Page SEO Twitter title 1`] = `"6 ตุลา : ธรรมศาสตร์จัดกิจกรรมรำลึก 43 ปี 6 ตุลา 2519 - BBC News ไทย"`;

exports[`AMP Photo Gallery Page Timestamp should match text and date 1`] = `
{
  "date": "2019-10-06",
  "text": "6 ตุลาคม 2019",
}
`;
